### PR TITLE
ci: clean up github actions workflows

### PR DESCRIPTION
**Pull request by supershal**
**What problem does this PR solve?**:
- use name for the actions, steps wherever it was missing
- use go version from `go.mod` file
- remove coverage.yaml workflow. merged it in the unit-tests.yaml workflow.
- merge golang-ci and lint workflow into one file
- run build, lint and test job on tags
- use d2iq docker mirror registry

### CODE REVIEW

**Review by dkoshkin — APPROVED** [review]
Thank you for the cleanup 🙏

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -6,22 +6,23 @@ on:
   push:
     branches:
       - main
+    tags:
+      - 'v*'
```
Comment (faiq) [inline .github/workflows/build.yml:10]: Why do we want to run on tags?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -4,27 +4,29 @@ name: Lint
 # yamllint disable-line rule:truthy
 on:
   push:
+    tags:
+      - 'v*'
```
Comment (faiq) [inline .github/workflows/lint.yml:8]: Again why do we need to run this on tags?

**Review by faiq — APPROVED** [review]
Some questions

File: .github/workflows/build.yml (hunk 1 above)
Comment (supershal) [reply .github/workflows/build.yml:10]: My initial thought was that if for any reason we have to manually force push a release tag, we should run basic workflows build, test and lint.
The current `make release` target does not cover all of them.

Comment (faiq) [reply .github/workflows/build.yml:10]: Gotcha. makes sense

**Pull request merged**